documentation-site/components/survey.js

```diff
@@ -0,0 +1,148 @@
+        paddingBottom: theme.sizing.scale800,
+        backgroundColor: theme.colors.background,
+        [`@media screen and (max-width: ${theme.breakpoints.medium}px`]: {
+          display: 'none',
+        },
+      })}
+    >
+      <div
+        className={useCss({
+          marginBottom: theme.sizing.scale400,
+        })}
+      >
+        <Label2>Overall, how satisfied are you with Base Web?</Label2>
+      </div>
+      <StarRating
+        numItems={5}
+        onChange={data => {
+          setRating(data.value);
+          trackEvent('survey', 'csat', data.value);
+          window.open(`
+            https://docs.google.com/forms/d/e/1FAIpQLSfQ6uxhW96LX31x5hbp2xIb-WPI0eIavfAb_s7lkhKnz-LIJQ/viewform?usp=pp_url&entry.1493106267=${data.value}
+          `);
+          setIsOpen(false);
+        }}
+        value={rating}
+      />
+      <div
+        role="button"
+        tabIndex="0"
+        onKeyPress={event => {
+          if (event.key === 'Enter') {
+            setIsOpen(false);
+          }
+        }}
+        className={useCss({
+          cursor: 'pointer',
+          position: 'absolute',
+          top: theme.sizing.scale400,
+          right: theme.sizing.scale400,
+        })}
+        onClick={() => {
+          setIsOpen(false);
+        }}
+      >
+        <Delete size="24px" />
+      </div>
+    </div>
+  ) : null;
+};
+
+// we want to show the survey once every two months
+function shouldShowSurvey() {
+  try {
+    // forcing the display of the survey for testing purposes
+    const {search} = window.location;
+    if (search && search.includes('survey')) {
+      return {showSurvey: true, delay: 0};
+    }
+  } catch (err) {
+    // do nothing
+  }
+
+  const Cookies = new CookiesConstructor();
+
+  const cookies = {
+    FIRST_SEEN: 'survey-user-first-seen',
+    LAST_SURVEYED: 'survey-last-surveyed',
+  };
+  const TWO_MONTHS_AGO = 60 * 24 * 60 * 60 * 1000;
+  const ONE_WEEK_AGO = 7 * 24 * 60 * 60 * 1000;
+  const SURVEY_DELAY = 2 * 60 * 1000; // 2 minutes
+
+  const firstSeen = Cookies.get(cookies.FIRST_SEEN);
+  const lastSurveyed = Cookies.get(cookies.LAST_SURVEYED);
+  const delay = 0;
+  const now = new Date();
+
+  // the person is a new visitor, bailing out
+  if (!firstSeen) {
+    Cookies.set(cookies.FIRST_SEEN, now.getTime(), {
+      // by default, cookies are session cookies, so without an expiration they are removed
+      // when the browser window is closed.
+      expires: new Date(now.getFullYear() + 10, now.getMonth()),
+    });
+    return {showSurvey: false, delay};
+  }
+
+  // the person only knows base web for less than a week
+  if (firstSeen > ONE_WEEK_AGO) {
+    return {showSurvey: false, delay};
+  }
+
+  // the person was surveyed less than two months ago, bailing out
+  if (lastSurveyed > TWO_MONTHS_AGO) {
+    return {showSurvey: false, delay};
+  }
+
+  Cookies.set(cookies.LAST_SURVEYED, now.getTime(), {
+    expires: new Date(now.getFullYear() + 10, now.getMonth()),
```

Comment:
is this correct? looks like we set expiration to 10 years in the future. should be be like below?
```suggestion
    expires: new Date(now.getFullYear(), now.getMonth() + 2),
```

Reply:
the idea was to set it for 10 years, as we use the value of this cookie to decide if we show it to the user - does that make sense?

also, if we'd like, we could set the expiration for 2 months, they way we'd check if the cookie value is undefined

Reply:
ha right, more than one datetime going on in the cookie